pkg/kubelet/nodestatus/setters_test.go

```diff
@@ -441,6 +454,31 @@ func TestNodeAddress(t *testing.T) {
 				"alpha.kubernetes.io/provided-node-ip": "10.1.1.1",
 			},
 			shouldError: false,
+			initializeNodeAddressesCloudProviderExternal: true,
+		},
+		{
+			name:                           "External cloud provider, node address is already set and initializeNodeAddressesCloudProviderExternal set",
+			nodeIP:                         netutils.ParseIPSloppy("10.1.1.1"),
+			cloudProviderType:              cloudProviderExternal,
+			nodeAddresses:                  []v1.NodeAddress{existingNodeAddress},
+			expectedAddresses:              []v1.NodeAddress{existingNodeAddress},
+			shouldError:                    true,
+			shouldSetNodeAddressBeforeTest: true,
+			initializeNodeAddressesCloudProviderExternal: true,
+		},
+		{
+			name:              "External cloud provider gets nodeIP annotation",
```

Comment:
All of the "with initializeNodeAddressesCloudProviderExternal set" tests are going to go away in the future, and the without initializeNodeAddressesCloudProviderExternal ones will be the default behavior, right?

So for git history preservation purposes, it seems like it would be better to leave the _existing_ test cases with (implicit) `initializeNodeAddressesCloudProviderExternal: false` and just remove the `expectedAddresses` from them, and then have all of the _new_ tests be the `initializeNodeAddressesCloudProviderExternal: true` ones copying over the old `expectedAddress` values from the existing tests.